cg/cli/workflow/balsamic/base.py

```diff
@@ -222,85 +222,68 @@ def config_case(
-    if normal_path:
-        command_str += f" --normal {normal_path}"
-    if umi:
-        command_str += f" --umi"
-    if quality_trim:
-        command_str += f" --quality-trim"
-    if adapter_trim:
-        command_str += f" --adapter-trim"
-    command = [f"bash -c 'source activate {conda_env}; balsamic"]
-    command_str += "'"  # add ending quote from above line
-    command.extend(command_str.split(" "))
-
-    if dry:
-        LOG.info(" ".join(command))
-        return SUCCESS
-
-    process = subprocess.run(" ".join(command), shell=True)
+    arguments = {
+        "tumor": tumor_path,
+        "normal": normal_path,
+        "case_id": case_id,
+        "output_config": f"{case_id}.json",
+        "quality_trim": quality_trim,
+        "adapter_trim": adapter_trim,
+        "umi": umi,
+        "umi_trim_length": umi_trim_length,
+        "panel_bed": target_bed,
+    }
+
+    process = context.obj["balsamic_api"].config_case(arguments=arguments, dry=dry)
     return process
 
 
 @balsamic.command()
-@click.option("-d", "--dry-run", "dry", is_flag=True, help="print command to console")
+@click.option("--dry-run", "dry", is_flag=True, help="print command to console")
 @click.option(
-    "-r", "--run-analysis", "run_analysis", is_flag=True, default=False, help="start analysis",
+    "-r", "--run-analysis", "run_analysis", is_flag=True, default=False, help="start analysis"
 )
 @click.option("--config", "config_path", required=False, help="Optional")
-@PRIORITY_OPTION
+@ANALYSIS_TYPE_OPTION
 @EMAIL_OPTION
+@PRIORITY_OPTION
 @click.argument("case_id")
 @click.pass_context
-def run(context, dry, run_analysis, config_path, priority, email, case_id):
-    """Generate a config for the case_id."""
+def run(context, dry, run_analysis, case_id, analysis_type, config_path, email, priority):
+    """Run BALSAMIC analysis for case_id."""
 
-    conda_env = context.obj["balsamic"]["conda_env"]
-    slurm_account = context.obj["balsamic"]["slurm"]["account"]
-    priority = priority if priority else context.obj["balsamic"]["slurm"]["qos"]
-    root_dir = Path(context.obj["balsamic"]["root"])
     if not config_path:
-        config_path = Path.joinpath(root_dir, case_id, case_id + ".json")
+        config_path = f'{context.obj["balsamic"]["root"]}/{case_id}/{case_id}.json'
 
     # Call Balsamic
-    command_str = f" run analysis --account {slurm_account} -s {config_path}"
-
-    if run_analysis:
-        command_str += " --run-analysis"
+    arguments = {
+        "case_id": case_id,
+        "analysis_type": analysis_type,
+        "sample_config": config_path,
+        "email": email,
+        "priority": priority,
+    }
+    process = context.obj["balsamic_api"].run_analysis(
+        arguments=arguments, run_analysis=run_analysis, dry=dry
+    )
+    return process
 
-    if email:
-        command_str += f" --mail-user {email}"
 
-    command_str += f" --qos {priority}"
+def get_priority_as_text(case_obj):
```

Comment:
```suggestion
def get_priority_as_text(case_obj: models.Family) -> str:
```
need to import `from cg.store import models`